hathor/transaction/transaction.py

```diff
@@ -271,27 +280,48 @@ def get_complete_token_info(self) -> dict[TokenUid, TokenInfo]:
     def _get_token_info_from_inputs(self) -> dict[TokenUid, TokenInfo]:
         """Sum up all tokens present in the inputs and their properties (amount, can_mint, can_melt)
         """
-        token_dict: dict[TokenUid, TokenInfo] = {}
-
-        default_info: TokenInfo = TokenInfo(0, False, False)
 
         # add HTR to token dict due to tx melting tokens: there might be an HTR output without any
         # input or authority. If we don't add it, an error will be raised when iterating through
         # the outputs of such tx (error: 'no token creation and no inputs for token 00')
-        token_dict[self._settings.HATHOR_TOKEN_UID] = TokenInfo(0, False, False)
+        token_dict: dict[TokenUid, TokenInfo] = {
+            self._settings.HATHOR_TOKEN_UID: TokenInfo(0, False, False, None, [], [])}
 
         for tx_input in self.inputs:
             spent_tx = self.get_spent_tx(tx_input)
             spent_output = spent_tx.outputs[tx_input.index]
 
             token_uid = spent_tx.get_token_uid(spent_output.get_token_index())
-            (amount, can_mint, can_melt) = token_dict.get(token_uid, default_info)
+            token_info_version: TokenInfoVersion | None = None
+
+            if token_uid != self._settings.HATHOR_TOKEN_UID:
+                assert self.storage is not None
+                from hathor.transaction.token_creation_tx import TokenCreationTransaction
+                token_creation_tx = self.storage.get_transaction(token_uid)
+                assert isinstance(token_creation_tx, TokenCreationTransaction)
```

Comment:
What if `token_uid` points to a non-token-creation vertex? This assert would simply crash the full node while the expected behavior would be to reject the transaction. One can exploit this assert to execute a DoS attack.